content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -154,9 +154,170 @@ TODO.
 
 ## Extensionless asset support
 
-<div class="release-feature-authors">authors: @TODO</div>
+<div class="release-feature-authors">authors: @bushrat011899</div>
 
-TODO.
+### Some background
+
+In Bevy 0.12 and earlier, all assets needed a file extension which uniquely mapped to a single asset type. Conceptually, this makes sense: a `.json` file should be loaded by the `JsonLoader` as a `JsonFile` asset.
+
+```rust
+let settings = asset_server.load("data/settings.json");
+```
+
+Peeking behind the curtain, this is achieved by passing the **path** to the [`AssetServer`], which selects an [`AssetLoader`] based on the file extension. This is possible because every [`AssetLoader`] is required to declare up-front what file types it supports.
+
+```rust
+impl AssetLoader for JsonAssetLoader {
+    /* snip */
+
+    fn extensions(&self) -> &[&str] {
+        &["json"]
+    }
+}
+```
+
+But, what happens when you have two different asset types use the same extension? Consider an `AudioSettings` asset and a `GraphicsSettings` asset: both will be stored as JSON files, but they need to parsed as different asset types.
+
+```rust
+impl AssetLoader for AudioSettingsAssetLoader {
+    type Asset = AudioSettings;
+
+    /* snip */
+
+    fn extensions(&self) -> &[&str] {
+        &["json"]
+    }
+}
+
+impl AssetLoader for GraphicsSettingsAssetLoader {
+    type Asset = GraphicsSettings;
+
+    /* snip */
+
+    fn extensions(&self) -> &[&str] {
+        &["json"]
+    }
+}
+```
+
+Previously, Bevy would pick one of the above asset loaders to be the canonical `json` loader based on registration order.
+
+```rust
+app.init_asset_loader::<AudioSettingsAssetLoader>();
+
+// All JSON files will be loaded by AudioSettingsAssetLoader
+
+app.init_asset_loader::<GraphicsSettingsAssetLoader>();
+
+// All JSON files will now be loaded by GraphicsSettingsAssetLoader
+```
+
+The workaround for this is to create a compound file extension for each unique asset type, such as `.audio.json` and `.graphics.json`.
+
+### Relaxing constraints
+
+But we can do better! In Bevy 0.13, the asset type can now used to infer the [`AssetLoader`].
+
+```rust
+// Uses AudioSettingsAssetLoader
+let audio_settings = asset_server.load("data/audio.json");
```

Comment:
This example feels a little weird since there isn't code showing where the AudioSettingsAssetLoader type is inferred from. Maybe have a resource they're inserted into that has the type?

Reply:
Agreed, I should've been more explicit here